scripts/unbind-and-delete-all-services.sh

```diff
@@ -0,0 +1,389 @@
+  string="${1}"
+  echo "ERROR: ${string}" >&2
+  return 1
+}
+
+#### limit orgs to test set in test mode
+get_orgs() {
+  local org_json
+  if [[ "${test_orgs}" != "" ]]; then
+    org_json=$(cf_curl "/v3/organizations?names=${test_orgs}")
+  else
+    org_json=$(cf_curl "/v3/organizations")
+  fi
+  get_resources "${org_json}"
+}
+
+get_spaces() {
+  local guid
+  guid="${1}"
+  local space_json
+  space_json=$(cf_curl "/v3/spaces?organization_guids=${guid}")
+  get_resources "${space_json}"
+}
+
+get_service_instances() {
+  local guid
+  guid="${1}"
+  local service_instance_json
+  service_instance_json=$(cf_curl "/v3/service_instances?space_guids=${guid}")
+  get_resources "${service_instance_json}"
+}
+
+get_service_credential_bindings() {
+  local guid
+  guid="${1}"
+  local service_credential_binding_json
+  service_credential_binding_json=$(cf_curl "/v3/service_credential_bindings?service_instance_guids=${guid}")
+  get_resources "${service_credential_binding_json}"
+}
+
+#### allow certain resources to be processed last (service-brokers)
+order_last() {
+  local last_resource
+  last_resource="${1}"
+  local resources
+  resources="${2}"
+  if grep "${last_resource}" < <(echo "${resources}") >/dev/null 2>&1; then
+    echo "${resources}" | grep -v "${last_resource}"
+    echo "${last_resource}"
+  else
+    echo "${resources}"
+  fi
+}
+
+get_resources() {
+  local json
+  json="${1}"
+  jq -c -r '.resources[]' < <(echo "${json}") 2>/dev/null
+}
+
+get_resource_value() {
+  local key
+  key="${1}"
+  local json
+  json="${2}"
+  jq -c -r ".${key}" < <(echo "${json}") 2>/dev/null
+}
+
+get_app_guid_for_binding() {
+  local json
+  json="${1}"
+  jq -c -r '.relationships.app.data.guid' < <(echo "${json}") 2>/dev/null
+}
+
+get_app_name_for_guid() {
+  local guid
+  local app_json
+  local app_resource
+  guid="${1}"
+  app_json=$(cf_curl "/v3/apps?guids=${guid}")
+  app_resource=$(get_resources "${app_json}")
+  get_resource_value "name" "${app_resource}"
+}
+
+get_cf_target() {
+  local org_name
+  local space_name
+  org_name="${1}"
+  space_name="${2}"
+  if ! cf target -o "${org_name}" -s "${space_name}" >/dev/null 2>&1; then
+    print_error "could not target org/space: ${org_name}/${space_name}"
+  fi
+}
+
+unbind_service() {
+  local org_name
+  local space_name
+  local app_name
+  local service_instance_name
+  org_name="${1}"
+  space_name="${2}"
+  app_name="${3}"
+  service_instance_name="${4}"
+  get_cf_target "${org_name}" "${space_name}"
+  if ! cf unbind-service -w "${app_name}" "${service_instance_name}" >/dev/null 2>&1; then
+    print_error "could not unbind app/service: ${app_name}/${service_instance_name}"
+  fi
+}
+
+delete_service() {
+  local org_name
+  local space_name
+  local service_instance_name
+  org_name="${1}"
+  space_name="${2}"
+  service_instance_name="${3}"
+  get_cf_target "${org_name}" "${space_name}"
+  if ! cf delete-service -w -f "${service_instance_name}" >/dev/null 2>&1; then
```

Comment:
This wait option will push pipeline time out. Consider running these in parallel.

Reply:
will consider in next iteration